UnknownHostException

```
[StatTransfer] SFTP settings in config.yml are most likely incorrect
com.jcraft.jsch.JSchException: java.net.UnknownHostException: sftp.scuby.online
       at com.jcraft.jsch.Util.createSocket(Util.java:349)
       at com.jcraft.jsch.Session.connect(Session.java:215)
       at com.jcraft.jsch.Session.connect(Session.java:183)
       at com.kyzegs.stattransfer.StatTransferUtil.transfer(StatTransferUtil.java:22)
       at com.kyzegs.stattransfer.StatTransfer$1.run(StatTransfer.java:39)
       at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:99)
       at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:468)
       at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1294)
       at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:377)
       at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1209)
       at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:997)
       at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:178)
       at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.net.UnknownHostException: sftp.scuby.online
       at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:220)
       at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
       at java.base/java.net.Socket.connect(Socket.java:609)
       at java.base/java.net.Socket.connect(Socket.java:558)
       at java.base/java.net.Socket.<init>(Socket.java:454)
       at java.base/java.net.Socket.<init>(Socket.java:231)
       at com.jcraft.jsch.Util.createSocket(Util.java:343)
       ... 12 more
```

--- a/README.md
+++ b/README.md
@@ -5,6 +5,7 @@ This plugin does no actual stat tracking on its own. Instead, it siphons data fr
 ### config.yml
 ```yml
 language: english
+transfer-task-delay: 6000 # in server ticks. Default is 5 minutes
 sftp:
    host: localhost # The domain or IP that the plugin should attempt to upload to
    port: 22 # The port on which your SFTP server is listening
@@ -17,5 +18,6 @@ sftp:
 This plugin supports the following languages:
 
  - English - Original language
+ - Dutch - [Kyzegs](https://github.com/Kyzegs)
  - Bulgarian - [WolfNT90](https://github.com/WolfNT90)
  - Spanish - [nynxus](https://twitter.com/nynxus)

--- a/src/main/java/com/kyzegs/stattransfer/StatTransfer.java
+++ b/src/main/java/com/kyzegs/stattransfer/StatTransfer.java
@@ -6,14 +6,33 @@ import java.util.ResourceBundle;
 
 import org.bukkit.configuration.InvalidConfigurationException;
 import org.bukkit.plugin.java.JavaPlugin;
-import org.bukkit.scheduler.BukkitRunnable;
+import org.bukkit.scheduler.BukkitTask;
 
 import com.jcraft.jsch.JSchException;
 import com.jcraft.jsch.SftpException;
 
 public class StatTransfer extends JavaPlugin {
-	int sftpPort;
+	int sftpPort, taskDelay;
 	String sftpHost, sftpUser, sftpPassword, sftpRemoteDir;
+
+	private BukkitTask objBukkitTaskInstance;
+
+	private final Runnable objBukkitRunnableStatTransfer = new Runnable() {
+		@Override
+		public void run() {
+			try {
+				StatTransferUtil.transfer();
+			} catch (JSchException | SftpException e) {
+				getLogger().warning(getString("log.JSchSftpExceptionWarning"));
+				e.printStackTrace();
+			} finally {
+				objBukkitTaskInstance.cancel();
+				objBukkitTaskInstance = getServer().getScheduler()
+						.runTaskLater(StatTransfer.getPlugin(StatTransfer.class), objBukkitRunnableStatTransfer,
+								taskDelay);
+			}
+		}
+	};
 
 	@Override
 	public void onEnable() {
@@ -29,24 +48,24 @@ public class StatTransfer extends JavaPlugin {
 			e.printStackTrace();
 		}
 
-		sftpHost = getConfig().getString("sftp.host");
-		sftpPort = getConfig().getInt("sftp.port");
-		sftpUser = getConfig().getString("sftp.user");
-		sftpPassword = getConfig().getString("sftp.password");
-		sftpRemoteDir = getConfig().getString("sftp.remote-dir");
+		try {
+			taskDelay = getConfig().getInt("transfer-task-delay", 6000);
+			sftpHost = getConfig().getString("sftp.host");
+			sftpPort = getConfig().getInt("sftp.port");
+			sftpUser = getConfig().getString("sftp.user");
+			sftpPassword = getConfig().getString("sftp.password");
+			sftpRemoteDir = getConfig().getString("sftp.remote-dir");
+			objBukkitTaskInstance = getServer().getScheduler().runTaskLater(this, objBukkitRunnableStatTransfer,
+					taskDelay);
+		} catch (final Exception e) {
+			e.printStackTrace();
+			getPluginLoader().disablePlugin(this);
+		}
+	}
 
-		new BukkitRunnable() {
-			@Override
-			public void run() {
-				try {
-					StatTransferUtil.transfer();
-				} catch (JSchException | SftpException e) {
-					getLogger().warning(getString("log.JSchSftpExceptionWarning"));
-					e.printStackTrace();
-					cancel();
-				}
-			}
-		}.runTaskTimer(this, 0, 6000);
+	@Override
+	public void onDisable() {
+		getServer().getScheduler().cancelTasks(this);
 	}
 
 	public String getString(String key) {

--- a/src/main/resources/config.yml
+++ b/src/main/resources/config.yml
@@ -1,4 +1,5 @@
 language: english
+transfer-task-delay: 6000
 sftp:
   host: localhost
   port: 22

